test/test_replication.cpp

```diff
@@ -3733,6 +3792,47 @@ TEST(Replication_MoveSelectedLinkView)
 }
 
 
-} // anonymous namespace
+TEST(Replication_HistorySchemaVersion)
```

Comment:
I suggest also testing the normal path here so that we don't rely on `Upgrade_Database_2_3_Writes_New_File_Format` for this case (it is conditionally run).

```
TEST(Replication_HistorySchemaVersionNormal)
 {
     SHARED_GROUP_TEST_PATH(path);
     ReplSyncClient repl(path, 1);
     SharedGroup sg_1(repl);
     // it should be possible to have two open shared groups on the same thread
     // without any read/write transactions in between
     SharedGroup sg_2(repl);
 }
```